view/pe/teview.cpp

```diff
@@ -143,52 +143,56 @@ void TEView::CreateSections()
 	auto headerType = Type::StructureType(headerStruct);
 	QualifiedName headerName = string("TE_Header");
 	auto headerTypeId = Type::GenerateAutoTypeId("te", headerName);
 	QualifiedName headerTypeName = DefineType(headerTypeId, headerName, headerType);
 	DefineDataVariable(m_imageBase + m_headersOffset, Type::NamedType(this, headerTypeName));
+	DefineAutoSymbol(new Symbol(DataSymbol, "__te_header", m_imageBase + m_headersOffset, NoBinding));
 
 	StructureBuilder sectionBuilder;
-	sectionBuilder.AddMember(Type::IntegerType(8, false), "Name");
-	sectionBuilder.AddMember(Type::IntegerType(4, false), "VirtualSize");
-	sectionBuilder.AddMember(Type::IntegerType(4, false), "VirtualAddress");
-	sectionBuilder.AddMember(Type::IntegerType(4, false), "SizeOfRawData");
-	sectionBuilder.AddMember(Type::IntegerType(4, false), "PointerToRawData");
-	sectionBuilder.AddMember(Type::IntegerType(4, false), "PointerToRelocations");
-	sectionBuilder.AddMember(Type::IntegerType(4, false), "PointerToLinenumbers");
-	sectionBuilder.AddMember(Type::IntegerType(2, false), "NumberOfRelocations");
-	sectionBuilder.AddMember(Type::IntegerType(2, false), "NumberOfLinenumbers");
-	sectionBuilder.AddMember(Type::IntegerType(4, false), "Characteristics");
+	sectionBuilder.AddMember(Type::ArrayType(Type::IntegerType(1, true), 8), "name");
+	sectionBuilder.AddMember(Type::IntegerType(4, false), "virtualSize");
+	sectionBuilder.AddMember(Type::IntegerType(4, false), "virtualAddress");
+	sectionBuilder.AddMember(Type::IntegerType(4, false), "sizeOfRawData");
+	sectionBuilder.AddMember(Type::IntegerType(4, false), "pointerToRawData");
+	sectionBuilder.AddMember(Type::IntegerType(4, false), "pointerToRelocations");
+	sectionBuilder.AddMember(Type::IntegerType(4, false), "pointerToLineNumbers");
+	sectionBuilder.AddMember(Type::IntegerType(2, false), "numberOfRelocations");
+	sectionBuilder.AddMember(Type::IntegerType(2, false), "numberOfLineNumbers");
+	sectionBuilder.AddMember(Type::IntegerType(4, false), "characteristics");
+
 	auto sectionStruct = sectionBuilder.Finalize();
 	auto sectionType = Type::StructureType(sectionStruct);
-	for (size_t i = 0; i < m_sections.size(); i++)
-	{
-		QualifiedName sectionName = string("TE_Section_Header_") + to_string(i);
-		auto sectionTypeId = Type::GenerateAutoTypeId("te", sectionName);
-		QualifiedName sectionTypeName = DefineType(sectionTypeId, sectionName, sectionType);
-		DefineDataVariable(
-			m_imageBase + m_headersOffset + EFI_TE_IMAGE_HEADER_SIZE + (EFI_TE_SECTION_HEADER_SIZE * i),
-			Type::NamedType(this, sectionTypeName)
-		);
-	}
+	QualifiedName sectionName = string("TE_Section_Header");
+	auto sectionTypeId = Type::GenerateAutoTypeId("te", sectionName);
+	QualifiedName sectionTypeName = DefineType(sectionTypeId, sectionName, sectionType);
+	DefineDataVariable(
+		m_imageBase + m_headersOffset + EFI_TE_IMAGE_HEADER_SIZE,
+		Type::ArrayType(Type::NamedType(this, sectionTypeName), header.numberOfSections)
```

Comment:
`header` is not defined in this function. Use `m_sections.size()`.

Reply:
Done.